asapdiscovery-data/asapdiscovery/data/cli.py

```diff
@@ -2,181 +2,10 @@
-    type=str,
-    default="v1",
-    envvar="POSTERA_API_VERSION",
-    show_envvar=True,
-)
-@click.pass_context
-def postera(ctx, api_url, api_token, api_version):
-    ctx.ensure_object(dict)
-
-    ctx.obj["api_url"] = api_url
-    ctx.obj["api_token"] = api_token
-    ctx.obj["api_version"] = api_version
-
-
-@postera.group()
-@click.pass_context
-def moleculeset(ctx):
-    from .postera.molecule_set import MoleculeSetAPI
-
-    ctx.obj["moleculesetapi"] = MoleculeSetAPI(
-        ctx.obj["api_url"], ctx.obj["api_version"], ctx.obj["api_token"]
-    )
-
-
-@moleculeset.command(help="")
-@smiles_field_param
-@click.argument("molecule_set_name")
-@input_file_arg
-@click.pass_context
-def create(
-    ctx,
-    smiles_field,
-    molecule_set_name,
-    input_file,
-):
-    import pandas as pd
-
-    from .postera.molecule_set import MoleculeList
-
-    msa = ctx.obj["moleculesetapi"]
-    df = pd.read_csv(input_file)
-
-    if smiles_field not in df.columns:
-        raise ValueError(f"SMILES field '{smiles_field}' not found in input file")
-
-    molecule_list = MoleculeList.from_pandas_df(df, smiles_field=smiles_field)
-
-    molecule_set_id = msa.create(molecule_set_name, molecule_list)
-    click.echo(f"Created molecule set with id {molecule_set_id}")
-
-
-@moleculeset.command(help="")
-@click.pass_context
-def list(ctx):
-    msa = ctx.obj["moleculesetapi"]
-
-    click.echo(msa.list())
-
-
-@moleculeset.command(help="")
-@molecule_set_id_arg
-@click.pass_context
-def get(
-    ctx,
-    molecule_set_id,
-):
-    ...
-
-
-@moleculeset.command(help="")
-@molecule_set_id_arg
-@click.pass_context
-def get_molecules(
-    ctx,
-    molecule_set_id,
-):
-    msa = ctx.obj["moleculesetapi"]
-    df = msa.get_molecules(molecule_set_id)
-
-    outfile_name = f"MoleculeSet_{molecule_set_id}.csv"
-    df.to_csv(outfile_name, index=False)
-    click.echo(f"Wrote molecule set to {outfile_name}")
-
-
-@moleculeset.command(help="")
-@id_field_param
-@molecule_set_id_arg
-@input_file_arg
-@click.pass_context
-def add_molecules():
-    ...
-
-
-@moleculeset.command(help="")
-@id_field_param
-@molecule_set_id_arg
-@input_file_arg
-@click.pass_context
-def update_molecules(
-    ctx,
-    id_field,
-    molecule_set_id,
-    input_file,
-):
-    import pandas as pd
-
-    from .postera.molecule_set import MoleculeUpdateList
-
-    msa = ctx.obj["moleculesetapi"]
-
-    df = pd.read_csv(input_file)
-
-    if id_field not in df.columns:
-        raise ValueError(
-            f"PostEra molecule id field '{id_field}' not found in input file"
-        )
-
-    update_molecule_list = MoleculeUpdateList.from_pandas_df(df, id_field=id_field)
-
-    molecules_updated = msa.update_molecules(molecule_set_id, update_molecule_list)
-    click.echo(f"Updated molecules {molecules_updated} in set {molecule_set_id}")
+# import subcommands to register them with the cli
+from .aws.cli import aws  # noqa: F401, E402
+from .postera.cli import postera  # noqa: F401, E402
```

Comment:
If these can be moved to the top, we can drop the E402 noqa skip

Reply:
They cannot; that would yield a circular import. 

If we want to make flake8 happy here, we could create a `cli` subdirectory with both an `__init__.py` and a single submodule for the the `def cli` group. The imports of CLI subcommand groups along with the top-level could then live in the `__init__.py`. That seems a bit obtuse to me with little added value, but fine with doing it.

Reply:
No we don't need to fight the tools, using `noqa` is way better than jumping though too many hoops to make flak8 happy